plugins/org.genivi.faracon.cli/src/org/genivi/faracon/cli/Franca2AraConverter.xtend

```diff
@@ -53,21 +68,61 @@ class Franca2AraConverter extends AbstractFaraconConverter<FrancaModelContainer,
 		return francaModels.map [new FrancaModelContainer(it)].toList
 	}
 
+	override protected loadAllDeploymentSourceFiles(Collection<InputFile> deploymentInputFilePaths) {
+		FDeployStandaloneSetup.doSetup
+		val francaDeploymentModels = deploymentInputFilePaths.map [ deploymentInputFilePath |
+			// Load an input Franca deployment model.
+			val normalizedFrancaDeploymentFilePath = deploymentInputFilePath.absolutePath
+			getLogger().logInfo("Loading Franca deployment file " + normalizedFrancaDeploymentFilePath);
+			var FDModel francaDeploymentModel;
+			try {
+				francaDeploymentModel = francaDeploymentLoader.loadModel(normalizedFrancaDeploymentFilePath);
+				if (francaDeploymentModel === null) {
+					getLogger().logError("File " + normalizedFrancaDeploymentFilePath + " could not be loaded!");
+					return null
+				}
+				return francaDeploymentModel
+			} catch (Exception e) {
+				getLogger().logError("File " + normalizedFrancaDeploymentFilePath + " could not be loaded!");
+				return null
+			}
+		].filterNull.toList
+		// add all created franca models to the resource set of the converter
+		francaDeploymentModels.forEach [
+			resourceSet.resources.add(it.eResource)
+		]
+
+		// Register the deployment sections for interface types and type collections.
+		someipFrancaDeploymentData.clear
+		francaDeploymentModels.forEach [
+			val FDModelExtender fdmodelExt = new FDModelExtender(it)
+			for(FDInterface fdInterface : fdmodelExt.FDInterfaces) {
+				val FDeployedInterface deployedInterface = new FDeployedInterface(fdInterface)
+				someipFrancaDeploymentData.registerInterfaceDeployment(fdInterface.target, deployedInterface)
+			}
+			for(FDTypes fdTypes : fdmodelExt.FDTypesList) {
+				val FDeployedTypeCollection deployedTypeCollection = new FDeployedTypeCollection(fdTypes)
+				someipFrancaDeploymentData.registerTypeCollectionDeployment(fdTypes.target, deployedTypeCollection)
+			}
+		]
+	}
+
 	override protected transform(Collection<FrancaModelContainer> francaModelContainers) {
 		val allNonPrimitiveElementTypesOfAnonymousArrays = francaModelContainers.map [ francaModelContainer |
 			val francaModel = francaModelContainer.model
 			francaModel.eAllContents.filter(FTypedElement).filter[array && type?.derived !== null].map[type?.derived].toList
 		].flatten.toSet
 		araConnector.setAllNonPrimitiveElementTypesOfAnonymousArrays(allNonPrimitiveElementTypesOfAnonymousArrays)
-		
+
```

Comment:
I have not fully understood why we needs this clearPrimitiveTypesAnonymousArrays and where exactly this has to be done. It seems that the necessity to clean it on certiain places is not so nice as this could easily leed to errors once we need to extend the tool.

Reply:
The anonymous fixed sized array types for primitive types are created on demand during the transformation and are collected for a full conversion of a system (not only for one model). The purpose of 'clearPrimitiveTypesAnonymousArrays' is to setup this collection with a proper initial state.
Meanwhile, there are several steps that have to be done to prepare a full conversion. We should really refactor the tool implementation to handle this more nicely. Using the injection mechanism in a clever way (as you suggest below) might be a good approach to solve this.